### ISSUE TITLE
environments cannot be turned off after enabled

### ISSUE THREAD

**Issue by mhipszki** [eslint/eslint#3965]
**The version of ESLint you are using (run eslint -v)**

1.5.1

**What you did (the source code and ESLint configuration)**

source code

```js
/*eslint-env node: false, browser: true */

var a = window;
var b = document;
```

eslint config

```json
{
    "extends": "eslint:recommended",
    "env": { "node": true }
}
```

**The actual ESLint output complete with numbers**

```
  3:9  error  "window" is not defined          no-undef
  4:9  error  "document" is not defined        no-undef
```

**What you expected to happen instead**

Expected ESLint to remove all introduced globals by the `node` environment and add ones from the `browser` environment.

Apparently "disabling" an environment is not possible after enabling it in a subfolder via `.eslintrc` (or I missed something), neither inline in a single file.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by mysticatea** [comment]
Thanks for the report.

Yes, currently we cannot overwrite envs with `false`.
And I'm fixing now :)
Related: #3915, #3958.

**Comment by nzakas** [comment]
#3915 is the same, let's track there
